From 17278abdf52239d666d7a2f7fef10f92f12e2102 Mon Sep 17 00:00:00 2001
From: Bo Yang <bo.yang@amlogic.com>
Date: Tue, 14 Jul 2015 20:10:02 +0800
Subject: [PATCH 5701/5965] PD#108759: vout: change menu configuraiton

In the current code, CONFIG_AM_TV_OUTPUT not only select tvout,
but also select vout service & notify.

TV used vout service & notfiy, but not need the tvout.

We should to add a new configuration option to separate vout
service & notify from tvout.

The modification is discussed and decided by
	Evoke.Zhang
	Jets.Yan
	Jianfeng.Wang
	Sandy.Luo
	Jiaming.Huang
	Victor.Wan
	Bo.Yang

Rename CONFIG_AM_LCD_OUTPUT to CONFIG_AML_LCD_TABLET
Raname CONFIG_AM_TV_OUTPUT  to CONFIG_AML_TVOUT
Reanme CONFIG_AM_TV_OUTPUT2 to CONFIG_AML_TVOUT

Add CONFIG_AMLOGIC_VOUT and CONFIG_AMLOGIC_VOUT2

CONFIG_AMLOGIC_VOUT select vout and compile vout service
& notify by default.
CONFIG_AMLOGIC_VOUT2 select vout2 and compile vout2 service
& notify by default.

CONFIG_AML_TVOUT select tvout only.
CONFIG_AML_TVOUT2 select tvout2 only.

CONFIG_AML_LCD_TABLET select tablet lcd module.

The following configuration can be compiled successfully,
	meson6_defconfig
	meson6_tee_defconfig
	meson6tv_defconfig,
	meson6tvc_h15_defconfig
	meson6tvd_defconfig
	meson8_defconfig
	meson8_tee_defconfig
	meson8b_defconfig
	meson8b_tee_defconfig
	mesong9bb_defconfig
	mesong9tv_defconfig

The following configuration cannot be compiled before for
other reasons.
	meson8_vlsi_defconfig
	meson8b_vlsi_defconfig
	meson8m2_vlsi_defconfig

Change-Id: I7e509b3f9d74bc9cd04d442a78e26b674a7eda1f
---
 arch/arm/configs/meson6_defconfig             |  9 ++--
 arch/arm/configs/meson6_tee_defconfig         | 17 +++----
 arch/arm/configs/meson6tv_defconfig           |  4 +-
 arch/arm/configs/meson6tvc_h15_defconfig      |  4 +-
 arch/arm/configs/meson6tvd_defconfig          |  4 +-
 arch/arm/configs/meson8_defconfig             |  8 +++-
 arch/arm/configs/meson8_tee_defconfig         |  8 +++-
 arch/arm/configs/meson8_vlsi_defconfig        |  9 +++-
 arch/arm/configs/meson8b_defconfig            |  5 +++
 arch/arm/configs/meson8b_tee_defconfig        |  5 +++
 arch/arm/configs/meson8b_vlsi_defconfig       |  6 ++-
 arch/arm/configs/meson8m2_vlsi_defconfig      | 15 ++++---
 arch/arm/configs/mesong9bb_defconfig          |  3 +-
 arch/arm/configs/mesong9tv_defconfig          |  3 +-
 arch/arm/mach-meson6/Makefile                 |  4 +-
 arch/arm/mach-meson6/lcd/lcd_config.c         |  6 +--
 arch/arm/mach-meson8/Makefile                 |  2 +-
 arch/arm/mach-meson8/board-meson8-common.c    | 33 +++++++-------
 arch/arm/mach-meson8/lcd/lcd_config.c         |  2 +-
 arch/arm/mach-meson8b/Makefile                |  2 +-
 arch/arm/mach-meson8b/board-meson8b-common.c  | 33 +++++++-------
 arch/arm/mach-meson8b/lcd/lcd_config.c        |  2 +-
 arch/arm/mach-mesong9bb/Makefile              |  4 +-
 arch/arm/mach-mesong9bb/lcd/lcd_config.c      |  2 +-
 arch/arm/mach-mesong9tv/Makefile              |  4 +-
 arch/arm/mach-mesong9tv/lcd/lcd_config.c      |  2 +-
 drivers/amlogic/display/Makefile              | 24 +++++-----
 drivers/amlogic/display/vout/Kconfig          | 45 +++++++++++--------
 drivers/amlogic/display/vout/Makefile         | 20 +++------
 drivers/amlogic/display/vout/aml_lcd.dtd      |  2 +-
 drivers/amlogic/display/vout/aml_vout.dtd     |  2 +-
 .../amlogic/display/vout/lcd_extern/aml.dtd   |  2 +-
 drivers/amlogic/display/vout/lcdoutc.c        |  6 +--
 drivers/amlogic/display/vout/tvoutc.c         |  2 +-
 drivers/amlogic/display/vout2/Kconfig         | 21 +++++----
 drivers/amlogic/display/vout2/Makefile        |  9 ++--
 drivers/amlogic/display/vout2/aml.dtd         |  4 +-
 drivers/amlogic/display/vout2/tvregs.h        | 16 +++----
 drivers/amlogic/hdmi/hdmi_tx/hdmi_tx.c        | 16 +++----
 drivers/amlogic/hdmi/hdmi_tx_2/hdmi_tx_main.c | 16 +++----
 40 files changed, 212 insertions(+), 169 deletions(-)

diff --git a/arch/arm/configs/meson6_defconfig b/arch/arm/configs/meson6_defconfig
index 482a40fb1247..6243e95e56a5 100644
--- a/arch/arm/configs/meson6_defconfig
+++ b/arch/arm/configs/meson6_defconfig
@@ -227,15 +227,18 @@ CONFIG_RTL8192CU=m
 CONFIG_BT_DEVICE=y
 CONFIG_VSYNC_RDMA=y
 CONFIG_AM_VIDEO=y
-# CONFIG_AML_VOUT_FRAMERATE_AUTOMATION is not set
+CONFIG_AMLOGIC_VOUT=y
+CONFIG_AML_TVOUT=y
 CONFIG_AML_VDAC_HW_SWITCH=y
-CONFIG_AM_TV_OUTPUT2=y
+CONFIG_AML_LCD_TABLET=y
+CONFIG_AML_GAMMA_DEBUG=y
+CONFIG_AMLOGIC_VOUT2=y
+CONFIG_AML_TVOUT2=y
 CONFIG_FB_AM=y
 CONFIG_FB_OSD2_DEFAULT_WIDTH=32
 CONFIG_FB_OSD2_DEFAULT_HEIGHT=32
 CONFIG_FB_OSD2_DEFAULT_WIDTH_VIRTUAL=32
 CONFIG_FB_OSD2_DEFAULT_HEIGHT_VIRTUAL=32
-CONFIG_AM_FB_EXT=y
 CONFIG_AM_LOGO=y
 CONFIG_AM_HDMI_ONLY=y
 # CONFIG_TVIN_CSI is not set
diff --git a/arch/arm/configs/meson6_tee_defconfig b/arch/arm/configs/meson6_tee_defconfig
index 006694060ad8..1615cef251e7 100644
--- a/arch/arm/configs/meson6_tee_defconfig
+++ b/arch/arm/configs/meson6_tee_defconfig
@@ -168,7 +168,6 @@ CONFIG_AM_REMOTE=y
 CONFIG_ADC_KEYPADS_AM=y
 CONFIG_KEY_INPUT_CUSTOM_AM=y
 CONFIG_MESON_INPUT_TOUCHSCREEN=y
-CONFIG_AML_TOUCH_ALGORITHM_SUPPORT=m
 CONFIG_FOCALTECH_CAPACITIVE_TOUCHSCREEN=y
 CONFIG_GOODIX_GT81XX_CAPACITIVE_TOUCHSCREEN=y
 CONFIG_GSLX680_CAPACITIVE_TOUCHSCREEN=y
@@ -215,6 +214,7 @@ CONFIG_AML_NEXT_GEN_NAND=y
 CONFIG_AML_NAND_KEY=y
 # CONFIG_AM_NAND is not set
 CONFIG_AM_ETHERNET=y
+CONFIG_AM_PHY=y
 CONFIG_AML_LAN8720=y
 CONFIG_AML_IP101_PHY=y
 CONFIG_AML_KSZ8091=y
@@ -228,16 +228,18 @@ CONFIG_RTL8192CU=m
 CONFIG_BT_DEVICE=y
 CONFIG_VSYNC_RDMA=y
 CONFIG_AM_VIDEO=y
-CONFIG_AM_TV_OUTPUT2=y
-CONFIG_AML_VOUT_FRAMERATE_AUTOMATION=n
+CONFIG_AMLOGIC_VOUT=y
+CONFIG_AML_TVOUT=y
 CONFIG_AML_VDAC_HW_SWITCH=y
+CONFIG_AML_LCD_TABLET=y
+CONFIG_AML_GAMMA_DEBUG=y
+CONFIG_AMLOGIC_VOUT2=y
+CONFIG_AML_TVOUT2=y
 CONFIG_FB_AM=y
-CONFIG_FB_OSD2_CURSOR=n
 CONFIG_FB_OSD2_DEFAULT_WIDTH=32
 CONFIG_FB_OSD2_DEFAULT_HEIGHT=32
 CONFIG_FB_OSD2_DEFAULT_WIDTH_VIRTUAL=32
 CONFIG_FB_OSD2_DEFAULT_HEIGHT_VIRTUAL=32
-CONFIG_AM_FB_EXT=y
 CONFIG_AM_LOGO=y
 CONFIG_AM_HDMI_ONLY=y
 # CONFIG_TVIN_CSI is not set
@@ -257,7 +259,7 @@ CONFIG_V4L_AMLOGIC_VIDEO2=y
 CONFIG_AMLOGIC_IONVIDEO=y
 CONFIG_DEINTERLACE=y
 # CONFIG_CRYPTO_AML_HW_CRYPTO is not set
-# CONFIG_CRYPTO_DEVICE_DRIVER is not set
+CONFIG_AML_TOUCH_ALGORITHM_SUPPORT=m
 CONFIG_UEVENT_HELPER_PATH="/sbin/hotplug"
 CONFIG_DEVTMPFS=y
 CONFIG_DEVTMPFS_MOUNT=y
@@ -287,6 +289,7 @@ CONFIG_INPUT_EVDEV=y
 CONFIG_INPUT_MISC=y
 CONFIG_INPUT_UINPUT=y
 CONFIG_DEVPTS_MULTIPLE_INSTANCES=y
+CONFIG_GPIO_SYSFS=y
 CONFIG_MEDIA_CAMERA_SUPPORT=y
 CONFIG_MEDIA_USB_SUPPORT=y
 CONFIG_USB_VIDEO_CLASS=y
@@ -361,5 +364,3 @@ CONFIG_SECURITY_NETWORK=y
 CONFIG_SECURITY_SELINUX=y
 CONFIG_CRC_T10DIF=y
 CONFIG_CRC7=y
-CONFIG_GPIO_SYSFS=y
-CONFIG_AM_PHY=y
diff --git a/arch/arm/configs/meson6tv_defconfig b/arch/arm/configs/meson6tv_defconfig
index 94e13167703b..4e979e7a48a3 100644
--- a/arch/arm/configs/meson6tv_defconfig
+++ b/arch/arm/configs/meson6tv_defconfig
@@ -174,7 +174,9 @@ CONFIG_AMAUDIO2=y
 CONFIG_VSYNC_RDMA=y
 CONFIG_AM_VIDEO=y
 CONFIG_AM_VDEC_AVS=y
-# CONFIG_AM_LCD_OUTPUT is not set
+CONFIG_AMLOGIC_VOUT=y
+CONFIG_AML_TVOUT=y
+CONFIG_AML_VOUT_FRAMERATE_AUTOMATION=y
 CONFIG_AML_TV_LCD=y
 CONFIG_FB_AM=y
 CONFIG_FB_OSD1_DEFAULT_BITS_PER_PIXEL=32
diff --git a/arch/arm/configs/meson6tvc_h15_defconfig b/arch/arm/configs/meson6tvc_h15_defconfig
index 83e578edbe32..7a64df5d71fb 100644
--- a/arch/arm/configs/meson6tvc_h15_defconfig
+++ b/arch/arm/configs/meson6tvc_h15_defconfig
@@ -174,7 +174,9 @@ CONFIG_RTL8192CU=m
 CONFIG_AMAUDIO2=y
 CONFIG_VSYNC_RDMA=y
 CONFIG_AM_VIDEO=y
-# CONFIG_AM_LCD_OUTPUT is not set
+CONFIG_AMLOGIC_VOUT=y
+CONFIG_AML_TVOUT=y
+CONFIG_AML_VOUT_FRAMERATE_AUTOMATION=y
 CONFIG_AML_TV_LCD=y
 CONFIG_FB_AM=y
 CONFIG_FB_OSD1_DEFAULT_BITS_PER_PIXEL=32
diff --git a/arch/arm/configs/meson6tvd_defconfig b/arch/arm/configs/meson6tvd_defconfig
index 466f5e8bb850..f7d11875ed8e 100644
--- a/arch/arm/configs/meson6tvd_defconfig
+++ b/arch/arm/configs/meson6tvd_defconfig
@@ -173,7 +173,9 @@ CONFIG_AMAUDIO2=y
 CONFIG_VSYNC_RDMA=y
 CONFIG_AM_VIDEO=y
 CONFIG_AM_ENCODER=y
-# CONFIG_AM_LCD_OUTPUT is not set
+CONFIG_AMLOGIC_VOUT=y
+CONFIG_AML_TVOUT=y
+CONFIG_AML_VOUT_FRAMERATE_AUTOMATION=y
 CONFIG_AML_TV_LCD=y
 CONFIG_FB_AM=y
 CONFIG_FB_OSD1_DEFAULT_BITS_PER_PIXEL=32
diff --git a/arch/arm/configs/meson8_defconfig b/arch/arm/configs/meson8_defconfig
index 5e9ebcda85ed..8445070e0a2b 100644
--- a/arch/arm/configs/meson8_defconfig
+++ b/arch/arm/configs/meson8_defconfig
@@ -246,7 +246,13 @@ CONFIG_VSYNC_RDMA=y
 CONFIG_AM_VIDEO2=y
 CONFIG_AM_ENCODER=y
 CONFIG_AM_JPEG_ENCODER=y
-CONFIG_AM_TV_OUTPUT2=y
+CONFIG_AMLOGIC_VOUT=y
+CONFIG_AML_TVOUT=y
+CONFIG_AML_VOUT_FRAMERATE_AUTOMATION=y
+CONFIG_AML_LCD_TABLET=y
+CONFIG_AML_GAMMA_DEBUG=y
+CONFIG_AMLOGIC_VOUT2=y
+CONFIG_AML_TVOUT2=y
 CONFIG_FB_AM=y
 CONFIG_FB_OSD2_CURSOR=y
 CONFIG_FB_OSD2_DEFAULT_WIDTH=32
diff --git a/arch/arm/configs/meson8_tee_defconfig b/arch/arm/configs/meson8_tee_defconfig
index 6c8412198813..d9d50b97d55c 100644
--- a/arch/arm/configs/meson8_tee_defconfig
+++ b/arch/arm/configs/meson8_tee_defconfig
@@ -248,7 +248,13 @@ CONFIG_TVIN_VIUIN=y
 CONFIG_AM_VIDEO2=y
 CONFIG_AM_ENCODER=y
 CONFIG_AM_JPEG_ENCODER=y
-CONFIG_AM_TV_OUTPUT2=y
+CONFIG_AMLOGIC_VOUT=y
+CONFIG_AML_TVOUT=y
+CONFIG_AML_VOUT_FRAMERATE_AUTOMATION=y
+CONFIG_AML_LCD_TABLET=y
+CONFIG_AML_GAMMA_DEBUG=y
+CONFIG_AMLOGIC_VOUT2=y
+CONFIG_AML_TVOUT2=y
 CONFIG_FB_AM=y
 CONFIG_FB_OSD2_CURSOR=y
 CONFIG_FB_OSD2_DEFAULT_WIDTH=32
diff --git a/arch/arm/configs/meson8_vlsi_defconfig b/arch/arm/configs/meson8_vlsi_defconfig
index 34f94e3d1f8e..ca2c5a7c67eb 100644
--- a/arch/arm/configs/meson8_vlsi_defconfig
+++ b/arch/arm/configs/meson8_vlsi_defconfig
@@ -167,7 +167,13 @@ CONFIG_PHYLIB=y
 CONFIG_VSYNC_RDMA=y
 CONFIG_AM_VIDEO=y
 # CONFIG_AM_VDEC_H264_4K2K is not set
-CONFIG_AM_TV_OUTPUT2=y
+CONFIG_AMLOGIC_VOUT=y
+CONFIG_AML_TVOUT=y
+CONFIG_AML_VOUT_FRAMERATE_AUTOMATION=y
+CONFIG_AML_LCD_TABLET=y
+CONFIG_AML_GAMMA_DEBUG=y
+CONFIG_AMLOGIC_VOUT2=y
+CONFIG_AML_TVOUT2=y
 CONFIG_FB_AM=y
 CONFIG_FB_OSD2_CURSOR=y
 CONFIG_AM_FB_EXT=y
@@ -214,6 +220,7 @@ CONFIG_INPUT_UINPUT=y
 CONFIG_HW_RANDOM=y
 # CONFIG_HWMON is not set
 CONFIG_MEDIA_SUPPORT=y
+CONFIG_BACKLIGHT_LCD_SUPPORT=y
 CONFIG_SOUND=y
 CONFIG_SND=y
 CONFIG_SND_SOC=y
diff --git a/arch/arm/configs/meson8b_defconfig b/arch/arm/configs/meson8b_defconfig
index 098c968c6b9a..ac592a35ceff 100644
--- a/arch/arm/configs/meson8b_defconfig
+++ b/arch/arm/configs/meson8b_defconfig
@@ -242,6 +242,11 @@ CONFIG_VSYNC_RDMA=y
 CONFIG_AM_VIDEO=y
 CONFIG_AM_ENCODER=y
 CONFIG_AM_JPEG_ENCODER=y
+CONFIG_AMLOGIC_VOUT=y
+CONFIG_AML_TVOUT=y
+CONFIG_AML_VOUT_FRAMERATE_AUTOMATION=y
+CONFIG_AML_LCD_TABLET=y
+CONFIG_AML_GAMMA_DEBUG=y
 CONFIG_FB_AM=y
 CONFIG_FB_OSD2_CURSOR=y
 CONFIG_FB_OSD2_DEFAULT_WIDTH=32
diff --git a/arch/arm/configs/meson8b_tee_defconfig b/arch/arm/configs/meson8b_tee_defconfig
index 91af6ea72907..b6209f0bb61a 100644
--- a/arch/arm/configs/meson8b_tee_defconfig
+++ b/arch/arm/configs/meson8b_tee_defconfig
@@ -243,6 +243,11 @@ CONFIG_VSYNC_RDMA=y
 CONFIG_AM_VIDEO=y
 CONFIG_AM_ENCODER=y
 CONFIG_AM_JPEG_ENCODER=y
+CONFIG_AMLOGIC_VOUT=y
+CONFIG_AML_TVOUT=y
+CONFIG_AML_VOUT_FRAMERATE_AUTOMATION=y
+CONFIG_AML_LCD_TABLET=y
+CONFIG_AML_GAMMA_DEBUG=y
 CONFIG_FB_AM=y
 CONFIG_FB_OSD2_CURSOR=y
 CONFIG_FB_OSD2_DEFAULT_WIDTH=32
diff --git a/arch/arm/configs/meson8b_vlsi_defconfig b/arch/arm/configs/meson8b_vlsi_defconfig
index 9712b91f8b59..e658250843f5 100644
--- a/arch/arm/configs/meson8b_vlsi_defconfig
+++ b/arch/arm/configs/meson8b_vlsi_defconfig
@@ -165,7 +165,9 @@ CONFIG_MMC_AML=y
 # CONFIG_AML_NEXT_NAND is not set
 # CONFIG_AMAUDIO is not set
 CONFIG_AM_VIDEO=y
-# CONFIG_AM_LCD_OUTPUT is not set
+CONFIG_AMLOGIC_VOUT=y
+CONFIG_AML_TVOUT=y
+CONFIG_AML_VOUT_FRAMERATE_AUTOMATION=y
 CONFIG_FB_AM=y
 # CONFIG_AMLOGIC_BACKLIGHT is not set
 CONFIG_AM_GE2D=y
@@ -173,7 +175,6 @@ CONFIG_AM_LOGO=y
 CONFIG_POST_PROCESS_MANAGER=y
 CONFIG_POST_PROCESS_MANAGER_PPSCALER=y
 # CONFIG_CRYPTO_AML_HW_CRYPTO is not set
-# CONFIG_CRYPTO_DEVICE_DRIVER is not set
 CONFIG_UEVENT_HELPER_PATH="/sbin/hotplug"
 CONFIG_DEVTMPFS=y
 CONFIG_DEVTMPFS_MOUNT=y
@@ -207,6 +208,7 @@ CONFIG_INPUT_UINPUT=y
 CONFIG_HW_RANDOM=y
 CONFIG_POWER_SUPPLY=y
 CONFIG_MEDIA_SUPPORT=y
+CONFIG_MEDIA_CAMERA_SUPPORT=y
 CONFIG_SOUND=y
 CONFIG_SND=y
 CONFIG_SND_SOC=y
diff --git a/arch/arm/configs/meson8m2_vlsi_defconfig b/arch/arm/configs/meson8m2_vlsi_defconfig
index 0f5c58ed727b..dbeb78aff4e4 100644
--- a/arch/arm/configs/meson8m2_vlsi_defconfig
+++ b/arch/arm/configs/meson8m2_vlsi_defconfig
@@ -165,12 +165,6 @@ CONFIG_AML_GPIO_KEY=y
 # CONFIG_AML_DVFS is not set
 CONFIG_EFUSE=y
 CONFIG_AML_PLATFORM_THERMAL=y
-CONFIG_AML_VIRTUAL_THERMAL=y
-CONFIG_THERMAL=y
-CONFIG_CPU_THERMAL=y
-CONFIG_GPU_THERMAL=y
-CONFIG_CPUCORE_THERMAL=y
-CONFIG_GPUCORE_THERMAL=y
 CONFIG_MMC_AML=y
 # CONFIG_AML_NFTL_NEW is not set
 # CONFIG_AM_NAND is not set
@@ -178,7 +172,9 @@ CONFIG_MMC_AML=y
 # CONFIG_AMAUDIO is not set
 CONFIG_VSYNC_RDMA=y
 CONFIG_AM_VIDEO=y
-# CONFIG_AM_LCD_OUTPUT is not set
+CONFIG_AMLOGIC_VOUT=y
+CONFIG_AML_TVOUT=y
+CONFIG_AML_VOUT_FRAMERATE_AUTOMATION=y
 CONFIG_FB_AM=y
 # CONFIG_AMLOGIC_BACKLIGHT is not set
 CONFIG_AM_GE2D=y
@@ -225,6 +221,11 @@ CONFIG_INPUT_UINPUT=y
 CONFIG_HW_RANDOM=y
 CONFIG_SPI=y
 CONFIG_POWER_SUPPLY=y
+CONFIG_THERMAL=y
+CONFIG_CPU_THERMAL=y
+CONFIG_CPUCORE_THERMAL=y
+CONFIG_GPU_THERMAL=y
+CONFIG_GPUCORE_THERMAL=y
 CONFIG_REGULATOR=y
 CONFIG_MEDIA_SUPPORT=y
 # CONFIG_MEDIA_DIGITAL_TV_SUPPORT is not set
diff --git a/arch/arm/configs/mesong9bb_defconfig b/arch/arm/configs/mesong9bb_defconfig
index 7895056316a4..0b5da1552de6 100644
--- a/arch/arm/configs/mesong9bb_defconfig
+++ b/arch/arm/configs/mesong9bb_defconfig
@@ -183,8 +183,7 @@ CONFIG_BT_RTKBTUSB=m
 CONFIG_VSYNC_RDMA=y
 CONFIG_AM_VIDEO=y
 CONFIG_AM_ENCODER=y
-# CONFIG_AML_VOUT_FRAMERATE_AUTOMATION is not set
-# CONFIG_AM_LCD_OUTPUT is not set
+CONFIG_AMLOGIC_VOUT=y
 CONFIG_AML_TV_LCD_V2=y
 CONFIG_FB_AM=y
 CONFIG_FB_OSD1_DEFAULT_BITS_PER_PIXEL=32
diff --git a/arch/arm/configs/mesong9tv_defconfig b/arch/arm/configs/mesong9tv_defconfig
index d42263245046..ba70181d1b27 100644
--- a/arch/arm/configs/mesong9tv_defconfig
+++ b/arch/arm/configs/mesong9tv_defconfig
@@ -185,8 +185,7 @@ CONFIG_BT_RTKBTUSB=m
 CONFIG_VSYNC_RDMA=y
 CONFIG_AM_VIDEO=y
 CONFIG_AM_ENCODER=y
-# CONFIG_AML_VOUT_FRAMERATE_AUTOMATION is not set
-# CONFIG_AM_LCD_OUTPUT is not set
+CONFIG_AMLOGIC_VOUT=y
 CONFIG_AML_TV_LCD_V2=y
 CONFIG_FB_AM=y
 CONFIG_FB_OSD1_DEFAULT_BITS_PER_PIXEL=32
diff --git a/arch/arm/mach-meson6/Makefile b/arch/arm/mach-meson6/Makefile
index 5a7cea76a5aa..1ef1a17f157c 100644
--- a/arch/arm/mach-meson6/Makefile
+++ b/arch/arm/mach-meson6/Makefile
@@ -6,7 +6,7 @@
 #obj-y := clock.o time.o iomapping.o usbclock.o gpio.o mod_gate.o power_gate.o
 
 obj-y := cpu.o clock.o time.o iomapping.o usbclock.o  mod_gate.o power_gate.o gpio.o pinctrl.o
-#obj-y +=  usbclock.o usbsetting.o 
+#obj-y +=  usbclock.o usbsetting.o
 #obj-y += power_gate.o mod_gate.o watchdog.o
 obj-m :=
 obj-n :=
@@ -42,7 +42,7 @@ obj-$(CONFIG_CPU_IDLE) += cpuidle.o
 
 obj-$(CONFIG_AML_HDMI_TX) +=  hdmi_tx_hw/
 
-obj-$(CONFIG_AM_LCD_OUTPUT) +=  lcd/
+obj-$(CONFIG_AML_LCD_TABLET) +=  lcd/
 
 # ARM Trustzone
 obj-$(CONFIG_MESON_TRUSTZONE) += meson-smc.o meson-secure.o
diff --git a/arch/arm/mach-meson6/lcd/lcd_config.c b/arch/arm/mach-meson6/lcd/lcd_config.c
index acebe996cb2c..d8e8a56ef951 100755
--- a/arch/arm/mach-meson6/lcd/lcd_config.c
+++ b/arch/arm/mach-meson6/lcd/lcd_config.c
@@ -952,7 +952,7 @@ static void set_venc_ttl(Lcd_Config_t *pConf)
 {
     lcd_print("%s\n", __FUNCTION__);
     WRITE_LCD_REG(ENCT_VIDEO_EN,		0);
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
     if(pConf->lcd_misc_ctrl.vpp_sel)
         WRITE_LCD_REG_BITS(VPU_VIU_VENC_MUX_CTRL, 3, 2, 2);
     else
@@ -990,7 +990,7 @@ static void set_venc_lvds(Lcd_Config_t *pConf)
 	lcd_print("%s\n",__FUNCTION__);
 
 	WRITE_LCD_REG(ENCL_VIDEO_EN, 0);
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 	if	(pConf->lcd_misc_ctrl.vpp_sel) {
 		WRITE_LCD_REG_BITS(VPU_VIU_VENC_MUX_CTRL, 0, 2, 2); //viu2 select encl
 	}
@@ -1039,7 +1039,7 @@ static void set_venc_mlvds(Lcd_Config_t *pConf)
 
     WRITE_LCD_REG(ENCL_VIDEO_EN,           0);
 
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
     if(pConf->lcd_misc_ctrl.vpp_sel){
         WRITE_LCD_REG_BITS(VPU_VIU_VENC_MUX_CTRL, 0, 2, 2);	//viu2 select encl
     }
diff --git a/arch/arm/mach-meson8/Makefile b/arch/arm/mach-meson8/Makefile
index 4e4b80d40231..4902f53a13e2 100755
--- a/arch/arm/mach-meson8/Makefile
+++ b/arch/arm/mach-meson8/Makefile
@@ -43,7 +43,7 @@ obj-$(CONFIG_AML_PLATFORM_THERMAL) += thermal.o
 
 obj-$(CONFIG_AML_HDMI_TX) +=  hdmi_tx_hw/
 
-obj-$(CONFIG_AM_LCD_OUTPUT) +=  lcd/
+obj-$(CONFIG_AML_LCD_TABLET) +=  lcd/
 
 obj-$(CONFIG_GPIO_TEST) += gpiotest.o
 
diff --git a/arch/arm/mach-meson8/board-meson8-common.c b/arch/arm/mach-meson8/board-meson8-common.c
index d702b76e4632..88fed95d7e49 100755
--- a/arch/arm/mach-meson8/board-meson8-common.c
+++ b/arch/arm/mach-meson8/board-meson8-common.c
@@ -58,13 +58,13 @@ extern void __init meson_timer_init(void);
 static __init void meson8_reserve(void)
 {
 
-    /* 
-      * Reserved memory for hotplug: 
-      *   start address: PHYS_OFFSET, size 0x4000, 
-      *   toprevent other getting logical address 0xc0000000 and 
-      *   flushing valid data on "zero address"
-      */
-    memblock_reserve(PHYS_OFFSET,__pa(swapper_pg_dir) - PHYS_OFFSET);
+	/*
+	 *   Reserved memory for hotplug:
+	 *   start address: PHYS_OFFSET, size 0x4000,
+	 *   toprevent other getting logical address 0xc0000000 and
+	 *   flushing valid data on "zero address"
+	 */
+	memblock_reserve(PHYS_OFFSET,__pa(swapper_pg_dir) - PHYS_OFFSET);
 }
 
 __initdata struct map_desc meson_board_io_desc[1];
@@ -84,19 +84,22 @@ static void __init meson_map_io(void)
 }
 
 static struct of_device_id m8_of_platform_bus_ids[] = {
-		{.compatible = "simple-bus",},  
-		{},
+	{.compatible = "simple-bus",},
+	{},
 };
+
+#ifdef CONFIG_OF_LM
 static struct of_device_id m8_of_lm_bus_ids[] = {
-		{.compatible = "logicmodule-bus",},  
-		{},
+	{.compatible = "logicmodule-bus",},
+	{},
 };
+#endif
 
 static __init void meson_init_machine_devicetree(void)
 {
-	struct device *parent;	
+	struct device *parent;
 	parent = get_device(&platform_bus);
-	
+
 	of_platform_populate(NULL,m8_of_platform_bus_ids,NULL,parent);
 #ifdef CONFIG_OF_LM
 	of_lm_populate(NULL,m8_of_lm_bus_ids,NULL,NULL);
@@ -108,7 +111,7 @@ int meson_cache_of_init(void);
 static __init void meson_init_early(void)
 {
 	int rev;
-	
+
 	meson_cpu_version_init();
 	/*
 	 * Mali or some USB devices allocate their coherent buffers from atomic
@@ -137,7 +140,7 @@ static const char *m8_common_board_compat[] __initdata = {
 
 DT_MACHINE_START(AML8726_M8, "Amlogic Meson8")
 	.reserve	= meson8_reserve,
-//.nr_irqs	= 
+//.nr_irqs	=
 	.smp		= smp_ops(meson_smp_ops),
 	.map_io		= meson_map_io,/// dt - 1
 	.init_early	= meson_init_early,/// dt -2
diff --git a/arch/arm/mach-meson8/lcd/lcd_config.c b/arch/arm/mach-meson8/lcd/lcd_config.c
index 4c4213cd8ddb..f1fc1c93e7e8 100755
--- a/arch/arm/mach-meson8/lcd/lcd_config.c
+++ b/arch/arm/mach-meson8/lcd/lcd_config.c
@@ -726,7 +726,7 @@ static void set_venc_lcd(Lcd_Config_t *pConf)
 	lcd_print("%s\n",__FUNCTION__);
 
 	WRITE_LCD_REG(ENCL_VIDEO_EN, 0);
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 	if	(pConf->lcd_misc_ctrl.vpp_sel) {
 		WRITE_LCD_REG_BITS(VPU_VIU_VENC_MUX_CTRL, 0, 2, 2); //viu2 select encl
 	}
diff --git a/arch/arm/mach-meson8b/Makefile b/arch/arm/mach-meson8b/Makefile
index 33df5d839ca6..ab43f48b1226 100755
--- a/arch/arm/mach-meson8b/Makefile
+++ b/arch/arm/mach-meson8b/Makefile
@@ -42,7 +42,7 @@ obj-$(CONFIG_AML_PLATFORM_THERMAL) += thermal.o
 
 obj-$(CONFIG_AML_HDMI_TX) +=  hdmi_tx_hw/
 
-obj-$(CONFIG_AM_LCD_OUTPUT) +=  lcd/
+obj-$(CONFIG_AML_LCD_TABLET) +=  lcd/
 
 # ARM Trustzone
 obj-$(CONFIG_MESON_TRUSTZONE) += meson-smc.o meson-secure.o
diff --git a/arch/arm/mach-meson8b/board-meson8b-common.c b/arch/arm/mach-meson8b/board-meson8b-common.c
index a5597b136510..23248fa7d0e6 100755
--- a/arch/arm/mach-meson8b/board-meson8b-common.c
+++ b/arch/arm/mach-meson8b/board-meson8b-common.c
@@ -57,13 +57,13 @@ extern void __init meson_timer_init(void);
 static __init void meson8b_reserve(void)
 {
 
-    /* 
-      * Reserved memory for hotplug: 
-      *   start address: PHYS_OFFSET, size 0x4000, 
-      *   toprevent other getting logical address 0xc0000000 and 
-      *   flushing valid data on "zero address"
-      */
-    memblock_reserve(PHYS_OFFSET,__pa(swapper_pg_dir) - PHYS_OFFSET);
+	/*
+	 *   Reserved memory for hotplug:
+	 *   start address: PHYS_OFFSET, size 0x4000,
+	 *   toprevent other getting logical address 0xc0000000 and
+	 *   flushing valid data on "zero address"
+	 */
+	memblock_reserve(PHYS_OFFSET,__pa(swapper_pg_dir) - PHYS_OFFSET);
 }
 
 __initdata struct map_desc meson_board_io_desc[1];
@@ -83,19 +83,22 @@ static void __init meson_map_io(void)
 }
 
 static struct of_device_id m8_of_platform_bus_ids[] = {
-		{.compatible = "simple-bus",},  
-		{},
+	{.compatible = "simple-bus",},
+	{},
 };
+
+#ifdef CONFIG_OF_LM
 static struct of_device_id m8_of_lm_bus_ids[] = {
-		{.compatible = "logicmodule-bus",},  
-		{},
+	{.compatible = "logicmodule-bus",},
+	{},
 };
+#endif
 
 static __init void meson_init_machine_devicetree(void)
 {
-	struct device *parent;	
+	struct device *parent;
 	parent = get_device(&platform_bus);
-	
+
 	of_platform_populate(NULL,m8_of_platform_bus_ids,NULL,parent);
 #ifdef CONFIG_OF_LM
 	of_lm_populate(NULL,m8_of_lm_bus_ids,NULL,NULL);
@@ -107,7 +110,7 @@ int meson_cache_of_init(void);
 static __init void meson_init_early(void)
 {
 	int rev;
-	
+
 	meson_cpu_version_init();
 	/*
 	 * Mali or some USB devices allocate their coherent buffers from atomic
@@ -135,7 +138,7 @@ static const char *m8_common_board_compat[] __initdata = {
 
 DT_MACHINE_START(AML8726_M8, "Amlogic Meson8B")
 	.reserve	= meson8b_reserve,
-//.nr_irqs	= 
+//.nr_irqs	=
 	.smp		= smp_ops(meson_smp_ops),
 	.map_io		= meson_map_io,/// dt - 1
 	.init_early	= meson_init_early,/// dt -2
diff --git a/arch/arm/mach-meson8b/lcd/lcd_config.c b/arch/arm/mach-meson8b/lcd/lcd_config.c
index ba2cf112811a..63e5e4a5384b 100755
--- a/arch/arm/mach-meson8b/lcd/lcd_config.c
+++ b/arch/arm/mach-meson8b/lcd/lcd_config.c
@@ -593,7 +593,7 @@ static void set_venc_lcd(Lcd_Config_t *pConf)
 	lcd_print("%s\n",__FUNCTION__);
 
 	WRITE_LCD_REG(ENCL_VIDEO_EN, 0);
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 	if	(pConf->lcd_misc_ctrl.vpp_sel) {
 		WRITE_LCD_REG_BITS(VPU_VIU_VENC_MUX_CTRL, 0, 2, 2); //viu2 select encl
 	}
diff --git a/arch/arm/mach-mesong9bb/Makefile b/arch/arm/mach-mesong9bb/Makefile
index 9b629f47cccf..9c71ae7c2b8d 100644
--- a/arch/arm/mach-mesong9bb/Makefile
+++ b/arch/arm/mach-mesong9bb/Makefile
@@ -42,7 +42,7 @@ obj-$(CONFIG_AML_PLATFORM_THERMAL) += thermal.o
 
 obj-$(CONFIG_AML_HDMI_TX_20)   +=  hdmi_tx_hw_20/
 
-obj-$(CONFIG_AM_LCD_OUTPUT) +=  lcd/
+obj-$(CONFIG_AML_LCD_TABLET) +=  lcd/
 
 # ARM Trustzone
 obj-$(CONFIG_MESON_TRUSTZONE) += meson-smc.o meson-secure.o
@@ -51,4 +51,4 @@ plus_sec                      := $(call as-instr,.arch_extension sec,+sec)
 AFLAGS_meson-smc.o            := -Wa,-march=armv7-a$(plus_sec)
 endif
 
-obj-$(CONFIG_AM_TV_OUTPUT) += enc_clk_config.o
+obj-$(CONFIG_AMLOGIC_VOUT) += enc_clk_config.o
diff --git a/arch/arm/mach-mesong9bb/lcd/lcd_config.c b/arch/arm/mach-mesong9bb/lcd/lcd_config.c
index 05ca437817dc..55087abec0e1 100644
--- a/arch/arm/mach-mesong9bb/lcd/lcd_config.c
+++ b/arch/arm/mach-mesong9bb/lcd/lcd_config.c
@@ -521,7 +521,7 @@ static void set_venc_lcd(Lcd_Config_t *pconf)
 	lcd_print("%s\n", __func__);
 
 	WRITE_LCD_REG(ENCL_VIDEO_EN, 0);
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 	if (pconf->lcd_misc_ctrl.vpp_sel) {
 		/* viu2 select encl */
 		WRITE_LCD_REG_BITS(VPU_VIU_VENC_MUX_CTRL, 0, 2, 2);
diff --git a/arch/arm/mach-mesong9tv/Makefile b/arch/arm/mach-mesong9tv/Makefile
index 5fcd032f8792..7bb3fb11fc21 100644
--- a/arch/arm/mach-mesong9tv/Makefile
+++ b/arch/arm/mach-mesong9tv/Makefile
@@ -41,7 +41,7 @@ obj-$(CONFIG_AML_PLATFORM_THERMAL) += thermal.o
 
 obj-$(CONFIG_AML_HDMI_TX_20)   +=  hdmi_tx_hw_20/
 
-obj-$(CONFIG_AM_LCD_OUTPUT) +=  lcd/
+obj-$(CONFIG_AML_LCD_TABLET) +=  lcd/
 
 # ARM Trustzone
 obj-$(CONFIG_MESON_TRUSTZONE)	+= meson-smc.o meson-secure.o
@@ -50,4 +50,4 @@ plus_sec			:= $(call as-instr,.arch_extension sec,+sec)
 AFLAGS_meson-smc.o		:= -Wa,-march=armv7-a$(plus_sec)
 endif
 
-obj-$(CONFIG_AM_TV_OUTPUT) += enc_clk_config.o
+obj-$(CONFIG_AMLOGIC_VOUT) += enc_clk_config.o
diff --git a/arch/arm/mach-mesong9tv/lcd/lcd_config.c b/arch/arm/mach-mesong9tv/lcd/lcd_config.c
index 081f735005e9..f42a5dd436d5 100644
--- a/arch/arm/mach-mesong9tv/lcd/lcd_config.c
+++ b/arch/arm/mach-mesong9tv/lcd/lcd_config.c
@@ -521,7 +521,7 @@ static void set_venc_lcd(Lcd_Config_t *pconf)
 	lcd_print("%s\n", __func__);
 
 	WRITE_LCD_REG(ENCL_VIDEO_EN, 0);
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 	if (pconf->lcd_misc_ctrl.vpp_sel) {
 		/* viu2 select encl */
 		WRITE_LCD_REG_BITS(VPU_VIU_VENC_MUX_CTRL, 0, 2, 2);
diff --git a/drivers/amlogic/display/Makefile b/drivers/amlogic/display/Makefile
index 454fcfe0c36b..5bed5d3a281d 100644
--- a/drivers/amlogic/display/Makefile
+++ b/drivers/amlogic/display/Makefile
@@ -1,17 +1,13 @@
-
 # Makefile for osd and tv
-obj-$(CONFIG_AM_TV_OUTPUT)	+= vout/
-obj-$(CONFIG_AM_TCON_OUTPUT)	+= vout/
-obj-$(CONFIG_AM_GE2D)		+= ge2d/
-obj-$(CONFIG_FB_AM)		+= osd/
-obj-$(CONFIG_AM_FB_EXT)		+= osd_ext/
-obj-y						+= lcd/
-obj-$(CONFIG_AM_LOGO)		+= aml_logo/
-
-obj-y				+= backlight/
-obj-$(CONFIG_AMLOGIC_LED)	+= led/
-
-obj-$(CONFIG_AM_TV_OUTPUT2)	+= vout2/
-obj-$(CONFIG_AMLOGIC_SYS_LED)	+= sys_led/
+obj-$(CONFIG_AMLOGIC_VOUT)		+= vout/
+obj-$(CONFIG_AM_GE2D)			+= ge2d/
+obj-$(CONFIG_FB_AM)			+= osd/
+obj-$(CONFIG_AM_FB_EXT)			+= osd_ext/
+obj-y					+= lcd/
+obj-y					+= backlight/
+obj-$(CONFIG_AM_LOGO)			+= aml_logo/
+obj-$(CONFIG_AMLOGIC_VOUT2)		+= vout2/
+obj-$(CONFIG_AMLOGIC_SYS_LED)		+= sys_led/
+obj-$(CONFIG_AMLOGIC_LED)		+= led/
 
 
diff --git a/drivers/amlogic/display/vout/Kconfig b/drivers/amlogic/display/vout/Kconfig
index 9423846b6300..2e636bdc5cd5 100755
--- a/drivers/amlogic/display/vout/Kconfig
+++ b/drivers/amlogic/display/vout/Kconfig
@@ -1,39 +1,46 @@
 #
 # Video output configuration
 #
-menu "Amlogic video output module"
 
-config AM_TV_OUTPUT
-	tristate "TV output"
-	default y
-	help  
-		tv output module
+menuconfig AMLOGIC_VOUT
+	tristate "Amlogic video output support"
+	default n
+	help
+		Amlogic video output support
+
+if AMLOGIC_VOUT
+
+config AML_TVOUT
+	tristate "Video output and control module"
+	default n
+	help
+		Video output and control module support
 
 config AML_VOUT_FRAMERATE_AUTOMATION
 	bool "framerate automation"
-	default y
-	depends on AM_TV_OUTPUT
+	default n
+	depends on AML_TVOUT
 	help
 		framerate adjustment automatically from video source
 
 config AML_VDAC_HW_SWITCH
 	bool "vdac hardware switch"
 	default n
-	depends on AM_TV_OUTPUT
+	depends on AML_TVOUT
 	help
 		vdac hardware switch of cvbs/ypbpr for mx.
 
-config AM_LCD_OUTPUT
-	tristate "LCD output" 
-	default y
-	help  
-		LCD output module
+config AML_LCD_TABLET
+	tristate "Tablet LCD output module"
+	default n
+	help
+		Tablet LCD output module support
 
 config AML_GAMMA_DEBUG
-	tristate "AML Gamma debug"
-	default y
-	depends on  AM_LCD_OUTPUT
+	tristate "gamma debug"
+	default n
+	depends on  AML_LCD_TABLET
 	help
-		AML gamma debug
+		gamma debug support
 
-endmenu
+endif
diff --git a/drivers/amlogic/display/vout/Makefile b/drivers/amlogic/display/vout/Makefile
index 403cd2637a4d..22b9615f14b3 100755
--- a/drivers/amlogic/display/vout/Makefile
+++ b/drivers/amlogic/display/vout/Makefile
@@ -1,19 +1,11 @@
-obj-y 	+=display.o
-display-objs=vout_notify.o  vout_serve.o
+obj-y += vout_notify.o  vout_serve.o
 
-obj-$(CONFIG_AML_VDAC_HW_SWITCH)  += vdac_switch.o
+obj-$(CONFIG_AML_VDAC_HW_SWITCH)	+= vdac_switch.o
 
+obj-$(CONFIG_AML_TVOUT)				+= tvout.o
+tvout-objs := tvoutc.o tvconf.o enc_clk_config.o
 
-obj-$(CONFIG_AM_TV_OUTPUT)  += am_tvout.o
-am_tvout-objs :=   tvoutc.o  tvconf.o
-ifeq ($(CONFIG_ARCH_MESONG9TV),y)
-else
-ifeq ($(CONFIG_ARCH_MESONG9BB),y)
-else
-am_tvout-objs += enc_clk_config.o
-endif
-endif
-obj-$(CONFIG_AM_LCD_OUTPUT)  += lcdoutc.o
-obj-$(CONFIG_AM_LCD_OUTPUT)  += lcd_extern/
+obj-$(CONFIG_AML_LCD_TABLET)		+= lcdoutc.o
+obj-$(CONFIG_AML_LCD_TABLET)		+= lcd_extern/
 
 #EXTRA_CFLAGS += -DCONFIG_PM
diff --git a/drivers/amlogic/display/vout/aml_lcd.dtd b/drivers/amlogic/display/vout/aml_lcd.dtd
index 35828795daf1..b4947f48ebf7 100755
--- a/drivers/amlogic/display/vout/aml_lcd.dtd
+++ b/drivers/amlogic/display/vout/aml_lcd.dtd
@@ -1,4 +1,4 @@
-#ifdef CONFIG_AM_LCD_OUTPUT
+#ifdef CONFIG_AML_LCD_TABLET
 //$$ DEVICE="lcd"
 //$$ L2 PROP_STR = "status"
 
diff --git a/drivers/amlogic/display/vout/aml_vout.dtd b/drivers/amlogic/display/vout/aml_vout.dtd
index a536601a420c..04117daddc93 100755
--- a/drivers/amlogic/display/vout/aml_vout.dtd
+++ b/drivers/amlogic/display/vout/aml_vout.dtd
@@ -1,4 +1,4 @@
-#ifdef CONFIG_AM_TV_OUTPUT
+#ifdef CONFIG_AMLOGIC_VOUT
 //$$ DEVICE = "mesonvout"
 //$$ L2 PROP_STR = "status"
 	mesonvout{
diff --git a/drivers/amlogic/display/vout/lcd_extern/aml.dtd b/drivers/amlogic/display/vout/lcd_extern/aml.dtd
index bd39b51c35d8..2aa9205c6c8a 100755
--- a/drivers/amlogic/display/vout/lcd_extern/aml.dtd
+++ b/drivers/amlogic/display/vout/lcd_extern/aml.dtd
@@ -1,4 +1,4 @@
-#ifdef CONFIG_AM_LCD_OUTPUT
+#ifdef CONFIG_AML_LCD_TABLET
 //******************************************************************************
 //lcd extern 
 //Notice: only one dev support
diff --git a/drivers/amlogic/display/vout/lcdoutc.c b/drivers/amlogic/display/vout/lcdoutc.c
index 2551245dd719..fd08aff3e2ad 100644
--- a/drivers/amlogic/display/vout/lcdoutc.c
+++ b/drivers/amlogic/display/vout/lcdoutc.c
@@ -351,7 +351,7 @@ static int lcd_set_current_vmode(vmode_t mode)
     return 0;
 }
 
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 static int lcd_set_current_vmode2(vmode_t mode)
 {
     mutex_lock(&lcd_vout_mutex);
@@ -443,7 +443,7 @@ static vout_server_t lcd_vout_server={
     },
 };
 
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 static vout_server_t lcd_vout2_server={
     .name = "lcd_vout2_server",
     .op = {
@@ -478,7 +478,7 @@ static void _init_vout(void)
     //add lcd actual active area size
     printk("lcd actual active area size: %d %d (mm).\n", pDev->pConf->lcd_basic.h_active_area, pDev->pConf->lcd_basic.v_active_area);
     vout_register_server(&lcd_vout_server);
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
     vout2_register_server(&lcd_vout2_server);
 #endif
 }
diff --git a/drivers/amlogic/display/vout/tvoutc.c b/drivers/amlogic/display/vout/tvoutc.c
index d17608778d07..c7491fb19ef3 100755
--- a/drivers/amlogic/display/vout/tvoutc.c
+++ b/drivers/amlogic/display/vout/tvoutc.c
@@ -543,7 +543,7 @@ printk("%s[%d] mode is %d\n", __func__, __LINE__, mode);
     printk("%s[%d]\n", __func__, __LINE__);
     enable_vsync_interrupt();
 #endif
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 	switch(mode)
 	{
 		case TVMODE_480I:
diff --git a/drivers/amlogic/display/vout2/Kconfig b/drivers/amlogic/display/vout2/Kconfig
index 772ff840553b..75a118f6613c 100755
--- a/drivers/amlogic/display/vout2/Kconfig
+++ b/drivers/amlogic/display/vout2/Kconfig
@@ -1,14 +1,19 @@
 #
 # Video output configuration
 #
-menu "Amlogic video output2 module"
 
-config AM_TV_OUTPUT2
-	tristate "TV output2"
+menuconfig AMLOGIC_VOUT2
+	tristate "Amlogic video output2 support"
 	default n
-	help  
-		tv output2 module
+	help
+		Amlogic video output2 support
 
-		
-endmenu
-	
+if AMLOGIC_VOUT2
+
+config AML_TVOUT2
+	tristate "Video output2 and control module"
+	default n
+	help
+		Video output and control module support
+
+endif
diff --git a/drivers/amlogic/display/vout2/Makefile b/drivers/amlogic/display/vout2/Makefile
index f81652ed812b..92259790c385 100755
--- a/drivers/amlogic/display/vout2/Makefile
+++ b/drivers/amlogic/display/vout2/Makefile
@@ -1,7 +1,4 @@
-obj-y 	+=display2.o
-display2-objs=vout2_notify.o  vout2_serve.o
-
-
-obj-$(CONFIG_AM_TV_OUTPUT2)  += am_tvout2.o
-am_tvout2-objs :=   tvoutc2.o  tvconf2.o nulldisp.o
+obj-y += vout2_notify.o vout2_serve.o
 
+obj-$(CONFIG_AML_TVOUT2)		+= tvout2.o
+tvout2-objs := tvoutc2.o tvconf2.o nulldisp.o
diff --git a/drivers/amlogic/display/vout2/aml.dtd b/drivers/amlogic/display/vout2/aml.dtd
index 3d5ccfa378aa..ca060b7103b1 100755
--- a/drivers/amlogic/display/vout2/aml.dtd
+++ b/drivers/amlogic/display/vout2/aml.dtd
@@ -1,4 +1,4 @@
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 //$$ DEVICE = "mesonvout2"
 //$$ L2 PROP_STR = "status"
 	mesonvout2{
@@ -6,4 +6,4 @@
 		dev_name = "mesonvout2";
 		status = "okay";
 	};
-#endif
\ No newline at end of file
+#endif
diff --git a/drivers/amlogic/display/vout2/tvregs.h b/drivers/amlogic/display/vout2/tvregs.h
index e28e8c258ea3..f9de7a3fdeee 100755
--- a/drivers/amlogic/display/vout2/tvregs.h
+++ b/drivers/amlogic/display/vout2/tvregs.h
@@ -49,7 +49,7 @@ static const  reg_t tvreg_vclk_hd[]={
 
 static const  reg_t tvregs_720p[] = {
     {P_VENC_VDAC_SETTING,          0xff,  },
-#ifndef CONFIG_AM_TV_OUTPUT2
+#ifndef CONFIG_AMLOGIC_VOUT2
     {P_HHI_VID_CLK_CNTL,           0x0,},
 #endif
     {P_HHI_VID_PLL_CNTL2,          0x00040003},
@@ -57,7 +57,7 @@ static const  reg_t tvregs_720p[] = {
     {P_HHI_HDMI_AFC_CNTL,          0x8c0000c3},
     {P_HHI_VID_PLL_CNTL,           0x0001043e,},
     {P_HHI_VID_DIVIDER_CNTL,       0x00010843,},
-#ifndef CONFIG_AM_TV_OUTPUT2
+#ifndef CONFIG_AMLOGIC_VOUT2
     {P_HHI_VID_CLK_DIV,            0x100},
     {P_HHI_VID_CLK_CNTL,           0x80000,},
     {P_HHI_VID_CLK_CNTL,           0x88001,},
@@ -106,7 +106,7 @@ static const  reg_t tvregs_720p[] = {
     {P_VENC_VDAC_DACSEL3,          0x0001,},
     {P_VENC_VDAC_DACSEL4,          0x0001,},
     {P_VENC_VDAC_DACSEL5,          0x0001,},
-#ifndef CONFIG_AM_TV_OUTPUT2
+#ifndef CONFIG_AMLOGIC_VOUT2
     {P_VPU_VIU_VENC_MUX_CTRL,      0x000a,},
 #endif
     {P_VENC_VDAC_FIFO_CTRL,        0x1000,},
@@ -286,7 +286,7 @@ static const reg_t tvregs_480cvbs[] = {
 
 static const reg_t tvregs_480p[] = {
     {P_VENC_VDAC_SETTING,          0xff,  },
-#ifndef CONFIG_AM_TV_OUTPUT2
+#ifndef CONFIG_AMLOGIC_VOUT2
     {P_HHI_VID_CLK_CNTL,           0x0,       },
 #endif
     {P_HHI_VID_PLL_CNTL2,          0x00040003},
@@ -294,7 +294,7 @@ static const reg_t tvregs_480p[] = {
     {P_HHI_HDMI_AFC_CNTL,          0x8c0000c3},
     {P_HHI_VID_PLL_CNTL,           0x0001042d,},
     {P_HHI_VID_DIVIDER_CNTL,       0x00010843,},
-#ifndef CONFIG_AM_TV_OUTPUT2
+#ifndef CONFIG_AMLOGIC_VOUT2
     {P_HHI_VID_CLK_DIV,            0x100},
 
     {P_HHI_VID_CLK_CNTL,           0x88001,},
@@ -343,7 +343,7 @@ static const reg_t tvregs_480p[] = {
     {P_VENC_VDAC_DACSEL3,          0xf003,},
     {P_VENC_VDAC_DACSEL4,          0xf003,},
     {P_VENC_VDAC_DACSEL5,          0xf003,},
-#ifndef CONFIG_AM_TV_OUTPUT2
+#ifndef CONFIG_AMLOGIC_VOUT2
     {P_VPU_VIU_VENC_MUX_CTRL,      0x000a,},
 #endif
     {P_VENC_VDAC_FIFO_CTRL,        0x1fc0,},
@@ -609,7 +609,7 @@ static const reg_t tvregs_1080i_50hz[] = {
 static const reg_t tvregs_1080p[] = {
     {P_VENC_VDAC_SETTING,          0xff,  },
 //	{VCLK_HD},
-#ifndef CONFIG_AM_TV_OUTPUT2
+#ifndef CONFIG_AMLOGIC_VOUT2
     {P_HHI_VID_CLK_DIV,			 1		},
     {P_HHI_VID_CLK_CNTL,        	 0x0421,},
 #endif
@@ -638,7 +638,7 @@ static const reg_t tvregs_1080p[] = {
     {P_ENCP_VIDEO_VSO_BLINE,       0,     },
     {P_ENCP_VIDEO_VSO_ELINE,       5,     },
     {P_ENCP_VIDEO_MAX_LNCNT,       1124,  },
-#ifndef CONFIG_AM_TV_OUTPUT2
+#ifndef CONFIG_AMLOGIC_VOUT2
     {P_VPU_VIU_VENC_MUX_CTRL,      0x000a,},      //New Add. If not set, when system boots up, switch panel to HDMI 1080P, nothing on TV.
 #endif
     {P_VENC_VIDEO_PROG_MODE,       0x100, },
diff --git a/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx.c b/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx.c
index 393e21026e01..53051cbc8887 100644
--- a/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx.c
+++ b/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx.c
@@ -145,7 +145,7 @@ static int hpdmode = 1; /*
                             1, unmux hpd when unplug;
                             2, unmux hpd when unplug  or off;
                         */
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 static int force_vout_index = 0;
 #endif
 static int hdmi_prbs_mode = 0xffff; /* 0xffff=disable; 0=PRBS 11; 1=PRBS 15; 2=PRBS 7; 3=PRBS 31*/
@@ -192,7 +192,7 @@ return value: 1, vout; 2, vout2;
 */
 {
     int vout_index = 1;
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
     if(force_vout_index){
         vout_index = force_vout_index;
     }
@@ -216,7 +216,7 @@ return value: 1, vout; 2, vout2;
 const vinfo_t * hdmi_get_current_vinfo(void)
 {
     const vinfo_t *info;
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
     if(get_cur_vout_index() == 2){
         info = get_current_vinfo2();
         if(info == NULL){ //add to fix problem when dual display is not enabled in UI
@@ -1021,7 +1021,7 @@ static int hdmitx_notify_callback_v(struct notifier_block *block, unsigned long
     return 0;
 }
 
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 static int hdmitx_notify_callback_v2(struct notifier_block *block, unsigned long cmd , void *para)
 {
     if(get_cur_vout_index()!=2)
@@ -1049,7 +1049,7 @@ static struct notifier_block hdmitx_notifier_nb_v = {
     .notifier_call    = hdmitx_notify_callback_v,
 };
 
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 static struct notifier_block hdmitx_notifier_nb_v2 = {
     .notifier_call    = hdmitx_notify_callback_v2,
 };
@@ -1654,7 +1654,7 @@ static int amhdmitx_probe(struct platform_device *pdev)
         return r;
     }
     vout_register_client(&hdmitx_notifier_nb_v);
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
     vout2_register_client(&hdmitx_notifier_nb_v2);
 #endif
     aout_register_client(&hdmitx_notifier_nb_a);
@@ -1756,7 +1756,7 @@ static int amhdmitx_remove(struct platform_device *pdev)
     kthread_stop(hdmitx_device.task);
 
     vout_unregister_client(&hdmitx_notifier_nb_v);
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
     vout2_unregister_client(&hdmitx_notifier_nb_v2);
 #endif
     aout_unregister_client(&hdmitx_notifier_nb_a);
@@ -2013,7 +2013,7 @@ static  int __init hdmitx_boot_para_setup(char *s)
 
 __setup("hdmitx=",hdmitx_boot_para_setup);
 
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 MODULE_PARM_DESC(force_vout_index, "\n force_vout_index\n");
 module_param(force_vout_index, uint, 0664);
 #endif
diff --git a/drivers/amlogic/hdmi/hdmi_tx_2/hdmi_tx_main.c b/drivers/amlogic/hdmi/hdmi_tx_2/hdmi_tx_main.c
index 7f24db9b96fa..234a8711bc01 100644
--- a/drivers/amlogic/hdmi/hdmi_tx_2/hdmi_tx_main.c
+++ b/drivers/amlogic/hdmi/hdmi_tx_2/hdmi_tx_main.c
@@ -135,7 +135,7 @@ static int hpdmode = 1; /*
                             1, unmux hpd when unplug;
                             2, unmux hpd when unplug  or off;
                         */
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 static int force_vout_index = 0;
 #endif
 static int hdmi_prbs_mode = 0xffff; /* 0xffff=disable; 0=PRBS 11; 1=PRBS 15; 2=PRBS 7; 3=PRBS 31*/
@@ -182,7 +182,7 @@ return value: 1, vout; 2, vout2;
 */
 {
     int vout_index = 1;
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
     if(force_vout_index){
         vout_index = force_vout_index;
     }
@@ -206,7 +206,7 @@ return value: 1, vout; 2, vout2;
 const vinfo_t * hdmi_get_current_vinfo(void)
 {
     const vinfo_t *info;
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
     if(get_cur_vout_index() == 2){
         info = get_current_vinfo2();
         if(info == NULL){ //add to fix problem when dual display is not enabled in UI
@@ -966,7 +966,7 @@ static int hdmitx_notify_callback_v(struct notifier_block *block, unsigned long
     return 0;
 }
 
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 static int hdmitx_notify_callback_v2(struct notifier_block *block, unsigned long cmd , void *para)
 {
     if(get_cur_vout_index()!=2)
@@ -994,7 +994,7 @@ static struct notifier_block hdmitx_notifier_nb_v = {
     .notifier_call    = hdmitx_notify_callback_v,
 };
 
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 static struct notifier_block hdmitx_notifier_nb_v2 = {
     .notifier_call    = hdmitx_notify_callback_v2,
 };
@@ -1499,7 +1499,7 @@ static int amhdmitx_probe(struct platform_device *pdev)
         return r;
     }
     vout_register_client(&hdmitx_notifier_nb_v);
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
     vout2_register_client(&hdmitx_notifier_nb_v2);
 #endif
 #ifdef CONFIG_SND_SOC
@@ -1602,7 +1602,7 @@ static int amhdmitx_remove(struct platform_device *pdev)
     kthread_stop(hdmitx_device.task);
 
     vout_unregister_client(&hdmitx_notifier_nb_v);
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
     vout2_unregister_client(&hdmitx_notifier_nb_v2);
 #endif
 #ifdef CONFIG_SND_SOC
@@ -1851,7 +1851,7 @@ static  int __init hdmitx_boot_para_setup(char *s)
 
 __setup("hdmitx=",hdmitx_boot_para_setup);
 
-#ifdef CONFIG_AM_TV_OUTPUT2
+#ifdef CONFIG_AMLOGIC_VOUT2
 MODULE_PARM_DESC(force_vout_index, "\n force_vout_index\n");
 module_param(force_vout_index, uint, 0664);
 #endif
-- 
2.19.0

